hooks/path_info.py

```diff
@@ -24,18 +28,24 @@
 VERSION_REGEX = re.compile("(.*)([._-])v(\d+)\.?([^.]+)?$", re.IGNORECASE)
 
 # a regular expression used to extract the frame number from the file.
-# this implementation assumes the version number is of the form '.####'
+# this implementation assumes the frame number is of the form '.####'
 # coming just before the extension in the filename and just after a '.', '_',
 # or '-'.
 FRAME_REGEX = re.compile("(.*)([._-])(\d+)\.([^.]+)$", re.IGNORECASE)
 
+# a regular expression used to extrand the frame number AND version from the file.
+# this implementation assumes the frame number is of the form '.####'
+# coming just before the extension in the filename and just after a '.', '_',
+# or '-'.
+# also assuming the version number is of the form 'v###' coming before the frame number and frame seperator.
+VERSION_FRAME_REGEX = re.compile("(.*)([._-])v(.*)([._-])(\d+)\.([^.]+)$", re.IGNORECASE)
 
 class BasicPathInfo(HookBaseClass):
     """
     Methods for basic file path parsing.
     """
 
-    def get_publish_name(self, path, sequence=False):
+    def get_publish_name(self, path, sequence=False, version=False):
```

Comment:
suggest making the new arg `strip_frame=False` or something like that.

Reply:
True, my suggestion is ambiguous... but do you mean 'strip_version' in this case?... because currently, with sequence = False, the version is removed, but if sequence=True, then the version remains. So we need both strip_version and strip_frame I think... and to cover all bases you might as well add a strip_ext option too!

Reply:
My preference here would be to include the most common behaviors, keeping the interface fairly clean. Now that I'm looking at this again, maybe we should have just done something like this from the start: 

- if this is a sequence, remove the frame number
- try to remove any version number (acting on the result of ^ if applicable)

Maybe that would be sufficient and keep things simple (for most cases) and consistent. Anything beyond that would imply taking over the hook. Curious what you think.